umm-spec-lib/src/cmr/umm_spec/migration/doi_migration.clj

```diff
@@ -0,0 +1,35 @@
+(ns cmr.umm-spec.migration.doi-migration
+  "Contains helper functions for migrating between different versions of UMM DOI"
+  (:require
+   [cmr.common.util :as util :refer [update-in-each]]))
+
+(defn migrate-doi-up
+  "Migrate :DOI from CollectionCitation level up to collection level."
+  [c]
+  (if-let [doi-obj (some :DOI (:CollectionCitations c))]
+    (-> c
+      (update-in-each [:CollectionCitations] dissoc :DOI)
+      (assoc :DOI doi-obj))
+    c))
+
+(defn migrate-doi-down
+  "Migrate :DOI from collection level down to CollectionCitation level."
+  [c]
+  (if-let [doi-obj (:DOI c)]
+    (-> c
+      (update-in-each [:CollectionCitations] assoc :DOI doi-obj)
+      (dissoc :DOI))
+    c))
+
+(defn migrate-missing-reason-up
+  "Migrate nil :DOI to have MissingReason up."
+  [c]
+  (when-not (seq (:DOI c))
+    (assoc c :DOI {:MissingReason "Not Applicable"})))
+
+(defn migrate-missing-reason-down
+  "Migrate nil :DOI to have MissingReason down."
+  [c]
+  (if (seq (get-in c [:DOI :DOI]))
```

Comment:
Not sure if we need to call seq - I believe :DOI :DOI has to be a string.

Reply:
I believe the test here is for collection emptiness ...

Reply:
@chris-durbin oh, are you saying that at `[:DOI :DOI]` in the `c` data structure, the element there will never be a collection? It will always be a string?

Reply:
Right, I'm pretty sure that get-in should always return a string or nil.

Reply:
I went a little crazy with the seqs.